### ISSUE TITLE
[Bug?]: fs.read() reads zero length if yarn 2 or 3 installed, and not updated from classic.

### ISSUE THREAD

**Issue by Montaire** [yarnpkg/berry#3983]
### Self-service

- [X] I'd be willing to implement a fix

### Describe the bug

I ran into a really strange issue while I was installing freshly one of my projects.
`fs.read()` and `fs.readSync()` returns 0 byte, as if the file would be empty.
`fs.readFile()` and `fs.readFileSync()` is not effected.
But when I upgrade the project with the `yarn set version` command from classic the issue is not present. 

I am willing to implement a fix but for the moment being I don't know where to look, I am new to yarn.

### To reproduce

0. Install yarn (classic installed on my system)

#### Working
1. `yarn install`
2. `yarn set version berry` canary works as well
3. `yarn install`
4. `yarn start`

#### Not working
1. `yarn set version berry`
2. `yarn install`
3. `yarn start`

package.json
``` json
{
  "name": "yarn-error",
  "version": "1.0.0",
  "main": "index.js",
  "scripts": {
    "start": "node ./index.js"
  }
}
```
index.js

``` js
const fs = require("fs");

let fd = fs.openSync('./package.json', 'r');
let buffer = Buffer.alloc(128);
console.log(fd, fs.readSync(fd, buffer), buffer.toString())
fs.close(fd);

fs.open('./package.json', (err, fd) => {
    console.log(err);
    fs.read(fd, (error, length, buffer) => console.log(error, length, buffer.toString()));
});
```

### Environment

```shell
System:
    OS: Windows 10 10.0.19043
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
  Binaries:
    Node: 17.3.1 - ~\AppData\Local\Temp\xfs-8d5e6465\node.CMD
    Yarn: 3.1.1 - ~\AppData\Local\Temp\xfs-8d5e6465\yarn.CMD
    npm: 8.3.0 - C:\Program Files\nodejs\npm.CMD
```

**Comment by merceyz** [comment]
Released as `3.2.0-rc.11`
```sh
yarn set version canary && yarn
```

**Comment by Montaire** [comment]
Wow that was fast!
Thank you very much for the quick fix!
